runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java

```diff
@@ -621,25 +620,16 @@ private void takeScreenshot(String fileName, Image image)
 		// Draw the game onto the screenshot
 		graphics.drawImage(image, gameOffsetX, gameOffsetY, null);
 
-		File playerFolder;
-		if (client.getLocalPlayer() != null && client.getLocalPlayer().getName() != null)
+		File playerFolder = SCREENSHOT_DIR;
+		if (client.getUsername() != null)
 		{
-			final EnumSet<WorldType> worldTypes = client.getWorldType();
-			final boolean dmm = worldTypes.contains(WorldType.DEADMAN);
-			final boolean sdmm = worldTypes.contains(WorldType.SEASONAL_DEADMAN);
-			final boolean dmmt = worldTypes.contains(WorldType.DEADMAN_TOURNAMENT);
-			final boolean isDmmWorld = dmm || sdmm || dmmt;
-
-			String playerDir = client.getLocalPlayer().getName();
-			if (isDmmWorld)
+			String sanitizedUsername = client.getUsername().replaceAll("[\\W]", "_");
```

Comment:
Just use Text.standardize() and then .replace(' ', '_')

Reply:
✔️